doc/modules/grid_search.rst

```diff
@@ -167,6 +171,358 @@ variable that is log-uniformly distributed between ``1e0`` and ``1e3``::
+consistently ranked among the best candidates across all iterations. Each
+iteration is allocated an increasing amount of resources per candidate, here
+the number of samples.
+
+.. figure:: ../auto_examples/model_selection/images/sphx_glr_plot_successive_halving_iterations_001.png
+   :target: ../auto_examples/model_selection/plot_successive_halving_iterations.html
+   :align: center
+
+The ``ratio`` parameter controls the rate at which the resources grow, and
+the rate at which the number of candidates decreases. In each iteration, the
+number of resources per candidate is multiplied by ``ratio`` and the number
+of candidates is divided by the same ratio. Along with ``resource`` and
+``min_resources``, ``ratio`` is the most important parameter to control the
+search in our implementation. ``ratio`` effectively controls the number of
+iterations in :class:`HalvingGridSearchCV` and the number of candidates (if
+'auto') and iterations in :class:`HalvingRandomSearchCV`.
+``aggressive_elimination=True`` can also be used if the number of available
+resources is small. More control is available through tuning the
+``min_resources`` parameter. Each parameter and their interactions are
+described in more details below.
+
+These estimators are still **experimental**: their predictions
+and their API might change without any deprecation cycle. To use them, you
+need to explicitly import ``enable_successive_halving``::
+
+  >>> # explicitly require this experimental feature
+  >>> from sklearn.experimental import enable_successive_halving # noqa
+  >>> # now you can import normally from model_selection
+  >>> from sklearn.model_selection import HalvingGridSearchCV
+
+.. topic:: Examples:
+
+    * :ref:`sphx_glr_auto_examples_model_selection_plot_successive_halving_heatmap.py`
+    * :ref:`sphx_glr_auto_examples_model_selection_plot_successive_halving_iterations.py`
+
+Choosing ``min_resources`` and the number of candidates
+-------------------------------------------------------
+
+Beside ``ratio``, the two main parameters that influence the behaviour of a
+successive halving search are the ``min_resources`` parameter, and the
+number of candidates (or parameter combinations) that are evaluated.
+``min_resources`` is the amount of resources allocated at the first
+iteration for each candidate. The number of candidates is specified directly
+in :class:`HalvingRandomSearchCV`, and is determined from the ``param_grid``
+parameter of :class:`HalvingGridSearchCV`.
+
+Consider a case where the resource is the number of samples, and where we
+have 1000 samples. In theory, with ``min_resources=10`` and ``ratio=2``, we
+are able to run **at most** 7 iterations with the following number of
+samples: ``[10, 20, 40, 80, 160, 320, 640]``.
+
+But depending on the number of candidates, we might run less than 7
+iterations: if we start with a **small** number of candidates, the last
+iteration might use less than 640 samples, which is a waste of resources. For
```

Comment:
I wouldn't say it's a waste of resources. If it's iterations, it might be good to save iterations. But if we want to ensure we use the full dataset, then it might not be great.
So you could say "which means not using all the available resources (samples)"?